app/src/main/java/com/sun/note_01/untils/Constant.kt

```diff
@@ -1,4 +1,28 @@
 package com.sun.note_01.untils
 
+import java.util.regex.Pattern
+
 object Constant {
+
+    private val EMAIL_ADDRESS_PATTERN: Pattern = Pattern.compile(
+        "[a-zA-Z0-9\\+\\.\\_\\%\\-\\+]{1,256}" +
+                "\\@" +
+                "[a-zA-Z0-9][a-zA-Z0-9\\-]{0,64}" +
+                "(" +
+                "\\." +
+                "[a-zA-Z0-9][a-zA-Z0-9\\-]{0,25}" +
+                ")+"
+    )
+
+    private val PASSWORD_PATTERN: Pattern = Pattern.compile(
+        "^(?=.*[0-9])(?=.*[A-Z])(?=\\S+$).{8,}$"
+    )
+
+    fun checkEmail(email: String): Boolean {
+        return EMAIL_ADDRESS_PATTERN.matcher(email).matches()
+    }
+
+    fun checkPassword(email: String): Boolean {
+        return PASSWORD_PATTERN.matcher(email).matches()
+    }
```

Comment:
Việc check email chỉ có trong presenter sử dụng, không cần thiết để ra ngoài utils mất đi tính OOP
Cả lớp này có thể ghi trong presenter
```suggestion
    String.isValidEmail()(pattern: String = EMAIL_PATTERN): Boolean =  pattern.matcher(this).matches()
```